File: david/sessions/2018/medium/11_container_with_most_water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height) - 1
        best = float('-inf')

        while left < right:
            m = min(height[left], height[right])
            best = max(best, m * (right - left))
            if height[left] == m:
                left += 1
            else:
                right -= 1
        return best

File: pia/2020/medium/11_container_with_most_water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let max = 0;
  let i = 0;
  let j = height.length - 1;

  while (j > i) {
    const h1 = height[i];
    const h2 = height[j];
    const area = Math.min(h1, h2) * (j - i);

    max = Math.max(area, max);

    if (h2 > h1) {
      i++
    } else {
      j--;
    }
  }
  return max;
};
